File: xhrweb/ASL.Hrms.Web/ClientApp/src/app/shared/models/employee/employee-manager.ts
export class EmployeeManager {
  id?: any;
  employeeId: any;
  managerId?: any;
  managerName: string = "";
  isPrimaryManager: boolean=false;
  assignedBy?: any;
  assignDate?: any ;
  unAssignedBy?: any;
  unAssignDate?: any;
  isDeleted: boolean;
  companyId?: any ;
  managerTypeId?: any;
  managerTypeName: string = "";
  public constructor(init?: Partial<EmployeeManager>){
    Object.assign(this, init);
  }
}

File: xhrweb/TaskManagement.Core/Entities/EmployeeManager.cs
﻿using ASL.Hrms.SharedKernel;
using System;
using System.Collections.Generic;
using System.Text;

namespace TaskManagement.Core.Entities
{
    public class EmployeeManager : BaseEntity<Guid>
    {
        public Guid EmployeeId { get; private set; }
        public Guid? ManagerId { get; private set; }
        public Boolean IsPrimaryManager { get; private set; }
        public Guid? AssignedBy { get; private set; }
        public DateTime? AssignDate { get; private set; }
        public Guid? UnAssignedBy { get; private set; }
        public DateTime? UnAssignDate { get; private set; }
        public Boolean IsDeleted { get; private set; }
        public Guid CompanyId { get; private set; }
        public Guid? ManagerTypeId { get; private set; }


        public EmployeeManager(Guid id) : base(id) { }
        private EmployeeManager() : base(Guid.NewGuid()) { }
    }
}
